src/DTO/Address.php

```diff
@@ -6,6 +6,24 @@
 
 use JsonSerializable;
 
+/**
+ * @psalm-type ptAddress = array{
```

Comment:
Why `ptAddress` and not `Address`?
i.e. match class name? those are unique.

Reply:
@flumonion I renamed them to be slightly different to the class as psalm appears to fail when it is named the same, see below when running psalm:
```
ERROR: ImplementedReturnTypeMismatch - src/DTO/ChannelImage.php:56:16 - The inherited return type 'array<array-key, Stock2Shop\Share\DTO\Image>' 
for Stock2Shop\Share\DTO\Image::createArray is different to the implemented return type for Stock2Shop\Share\DTO\ChannelImage::createarray 'array<array-key, array{active: bool, channel_id: int, channel_image_code: string, delete: bool, id: int, success: bool}>' (see https://psalm.dev/123)
     * @return ChannelImage[]
```
The issue is related to the `@return` for the `createArray` function, psalm seems to get confused when the return type is the same as the psalm-type.

To resolve this I can either:
1. Use unique psalm-type names
2. Change the `@return` to be `self[]` on all the `createArray` functions

I am not sure which to suggest as I am not sure if using the same name as the class name may cause issues when the class is called in any other PHP prams docs

Reply:
`TypeAddress`, `TypeChannel` etc is fine...
`pt` is just a bit ambiguous :)